api/v1alpha2/kubeadmbootstrapconfig_types.go

```diff
@@ -36,6 +36,14 @@ type KubeadmConfigSpec struct {
 	// AdditionalUserDataFiles specifies extra files to be passed to user_data upon creation.
 	// +optional
 	AdditionalUserDataFiles []Files `json:"additionalUserDataFiles,omitempty"`
+	// AdditionalPreCommands specifies extra commands to be passed to user_data upon creation
+	// which run before kubeadm runs
+	// +optional
+	AdditionalPreCommands []string `json:"additionalPreCommands,omitempty"`
```

Comment:
What do we think about dropping the `Additional` prefix for these fields? Given that we have `Pre` and `Post` seems that should be fine?

Reply:
Interesting, i wonder why we had the word Additional commands in the first place? I guess to signify in addition to kubeadm init/join.

Does Additional add anything to this field? I think Additional adds a sense that these fields are optional. That if I don't need any additional commands I don't have to have them. But if they were just called Pre and Post I might feel more inclined like I need them.

Overall, I don't have any strong feelings and am happy with both Additional and Pre/Post. Up to @ykakarap if he'd like to update it or not.

Reply:
Additional signifies that it runs in addition to kubeadm init/join command as mentioned by @chuckha .

If we want to drop "Additional" I suggest naming them `PreKubeadmCommands` and `PostKubeadmCommands`. This will suggest that they are commands besides the kubeadm commands and are also more descriptive of their behavior.

Thoughts?

Reply:
That also works. I was almost worried about tying the name to an implementation and then remembered this is the kubeadm bootstrap provider :)

I have no opinion on this. I'm happy to merge with Additional or Pre/Post KubeadmCommands

Reply:
@vincepri do you have strong feelings here?

Reply:
I have not strong feeling, but If I have to express a preference is
- AdditionalFiles --> Files
- AdditionalPreCommands --> PreKubeadmCommands
- AdditionalPostCommands --> PostKubeadmCommands

Reply:
@fabriziopandini's suggestion looks good to me :)

Reply:
>AdditionalFiles --> Files

`AdditionalFiles` is in `BaseUserData` struct. Shall i only change that to `Files` or also change the `AdditionalUserDataFiles` to `UserDataFiles` in `KubeadmConfigSpec` ?

Reply:
The important change is the KubeadmConfigSpec.

Ideally we don't reference UserData anywhere as that is a legacy AWS thing, but we can do that in a different PR and not change the API.

Reply:
To be explicit, please change `AdditionalUserDataFiles` to `Files` on the `KubeadmConfigSpec` type

Reply:
Renamed the fields as discussed :)